app/routes.js

```diff
@@ -1,43 +1,119 @@
+                .findAllUsers(req)
                 .then(data => {
                     res.send(data);
-                }).catch(err => {
+                })
+                .catch(err => {
+                    res.status(500).send({
+                        message: err.message || "Unable to rerieve users"
+                    });
+                });
+        });
+
+
+        app.post("/api/v1/users/", (req, res) => {
+            return controller
+                .createUser(req.body)
+                .then(data => {
+                    res.send(data);
+                })
+                .catch(err => {
                     res.status(err.errorCode || 500).send({
                         message: err.message || "Some error occurred while creating the user."
                     });
                 });
+        });
+
+
+        app.get('/api/v1/users/:userId', (req, res) => {
+            return controller.findSingleUser(req)
+                .then(data => {
+                    if (!data) {
+                        return res.status(500).send({
+                            message: "There is no user with the ID of : " + req.params.userId
+                        })
+                    } else {
+                        console.log("Now displaying the info for the user with ID :" + req.params.userId)
+                    }
+                    res.send(data)
+                }).catch(err => {
+                    if (err.kind == 'ObjectId') {
+                        return res.status(404).send({
+                            error: "The ID entered does not match a valid user ID format. Please try entering a valid user ID."
+                        });
+                    }
+                })
         })
 
 
-        app.get('/api/v1/users/', (req, res) => {
-            return controller.findAllUsers(req)
+        app.put('/api/v1/users/:userId', (req, res) => {
+            return controller.findUserByIdAndUpdate(req)
                 .then(data => {
+                    if (!data) {
+                        return res.status(500).send({
+                            message: "There is no user with the ID of : " + req.params.userId
+                        })
+                    } else {
+                        console.log("Information for userID : " + req.params.userId + " has been edited.");
+                    }
                     res.send(data)
                 }).catch(err => {
-                    res.status(500).send({
-                        message: err.message || "Unable to rerieve users"
-                    })
+                    if (err) {
+                        return res.status(404).send({
+                            error: "There is no user with the ID of : " + req.params.userId
+                        });
+                    }
+                })
+        })
+
+
+        app.delete('/api/v1/users/:userId', (req, res) => {
+            return controller.findUserByIdAndDelete(req)
```

Comment:
Good naming here but be careful with functions that do more than one thing, in this case it's super clear that the function first looks for a user and then it deletes it...  the think is that most databases delete based on query directly.,, so probably you can name this functions as deleteById(userId)

Reply:
This has been renamed to deleteUserById